include/common/enum_errcode.hpp

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2019-2022 Second State INC
+
+//===-- wasmedge/common/enum_errcode.h - Error code C++ enumerations ------===//
+//
+// Part of the WasmEdge Project.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file contains the C++ enumerations of WasmEdge error code.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "dense_enum_map.h"
+#include "spare_enum_map.h"
+
+#include <cstdint>
+#include <string_view>
+
+namespace WasmEdge {
+
+/// WasmEdge runtime phasing C++ enumeration class.
+enum class WasmPhase : uint8_t {
+#define UseWasmPhase
+#define Line(NAME, VALUE, STRING) NAME = VALUE,
+#include "enum.inc"
+#undef Line
+#undef UseWasmPhase
+};
+
+static inline constexpr auto WasmPhaseStr = []() constexpr {
+  using namespace std::literals::string_view_literals;
+  std::pair<WasmPhase, std::string_view> Array[] = {
+#define UseWasmPhase
+#define Line(NAME, VALUE, STRING) {WasmPhase::NAME, STRING},
+#include "enum.inc"
+#undef Line
+#undef UseWasmPhase
+  };
+  return DenseEnumMap(Array);
+}
+();
+
+/// Error category C++ enumeration class.
+enum class ErrCategory : uint8_t {
+#define UseErrCategory
+#define Line(NAME, VALUE) NAME = VALUE,
+#include "enum.inc"
+#undef Line
+#undef UseErrCategory
+};
+
+class ErrCode {
+public:
+  /// Error code C++ enumeration class.
+  enum class Value : uint32_t {
```

Comment:
Use `enum` directly, so the `errors` can be exposed at the same layer.
Such as `ErrCode::XXXError`. It makes no sense to create a new layer called `Value`.

Reply:
This is not supported in dense_map.